packages/babel-helper-split-export-declaration/src/index.js

```diff
@@ -0,0 +1,65 @@
+import * as t from "@babel/types";
+
+const splitExportDeclaration = exportDeclaration => {
+  if (!exportDeclaration.isExportDeclaration()) {
+    throw new Error("Only export declarations can be splitted.");
+  }
+
+  // build specifiers that point back to this export declaration
+  const isDefault = exportDeclaration.isExportDefaultDeclaration();
+  const declaration = exportDeclaration.get("declaration");
+
+  if (isDefault) {
+    if (declaration.isIdentifier()) {
+      throw new Error("It doesn't make sense to split exported identifier.");
+    }
+    const standaloneDeclaration =
+      declaration.isFunctionDeclaration() || declaration.isClassDeclaration();
+
+    let id = declaration.node.id;
+    const liveBinding = !!id;
+
+    if (!id) {
+      id = declaration.scope.generateUidIdentifier("default");
+      if (standaloneDeclaration) {
+        declaration.node.id = t.cloneNode(id);
+      }
+    }
+
+    const updatedDeclaration = standaloneDeclaration
+      ? declaration
+      : t.variableDeclaration("var", [
+          t.variableDeclarator(t.cloneNode(id), declaration.node),
+        ]);
+
+    const updatedExportDeclaration = liveBinding
+      ? t.exportNamedDeclaration(null, [
+          t.exportSpecifier(t.cloneNode(id), t.identifier("default")),
+        ])
+      : t.exportDefaultDeclaration(t.cloneNode(id));
```

Comment:
Even if it doesn't have an ID, cyclic dependencies can still rely on it being hoisted. It's never safe to do `export default foo;` in this case.

Reply:
could you back it up with a sample code? not sure if I understand how the problem can be created